frontend/src/pages/groups/GroupForm.tsx

```diff
@@ -30,40 +32,53 @@ interface Props {
  * This component encapsulates a form for creating and editing groups.
  */
 const GroupForm: FunctionComponent<Props> = (props) => {
+  const { profile } = useContext(UserProfileContext)
+
   const {
     register,
     handleSubmit,
     reset,
     formState: { errors },
-  } = useForm()
+  } = useForm<GroupCreateInput>()
 
   useEffect(
     function resetFormValues() {
       if (props.defaultValues) {
         // Update the values of the fields
-        reset(props.defaultValues.group)
+        reset(_omit(props.defaultValues.group, ['id', '__typename']))
       }
     },
     [props.defaultValues, reset],
   )
 
+  const submitForm = handleSubmit((input) => {
+    // Non-admins can only create sending groups
+    if (!profile?.isAdmin) {
```

Comment:
Question (non-blocking): what trade-offs would there be to ensure that there is a profile available here and where other code that is uncertain about the presence of a profile runs? Just so we're not constantly using `?.`.  My concern is that we'll get accustomed to using `?.` so much we'll miss that someone actually missing a profile will get an experience they shouldn't.

Reply:
That's a great question and I'd like to do a little spike on it, so I opened #134.